candecode_node:
  ros__parameters:
    decode_choices: false
    warn_if_unknown: false
    dbc_file_S: /home/as/as_pipeline/ros2_candecode/config/RSDB_CAN_S_Process_Data.dbc
    dbc_file_M: /home/as/as_pipeline/ros2_candecode/config/RSDB_CAN_M.dbc
    imu_mapping:
      frame_id: "base_link"
      canbus_message_acc_id: 296
      canbus_message_gyro_id: 312
      field_mapping:
        acc:
          x: Gyro_Acceleration_X
          y: Gyro_Acceleration_Y
          z: Gyro_Acceleration_Z
        gyro:
          x: Gyro_YawRate_X
          y: Gyro_YawRate_Y
          z: Gyro_YawRate_Z
    steering_mapping:
      frame_id: "base_link"
      canbus_message_id: 515
      field_mapping:
        steering: NodeF_SteerAngle
    vsu_mapping:
      frame_id: "base_link"
      canbus_message_vsu_distance_id: 268
      canbus_message_vsu_angle_id: 284
      field_mapping:
        distance: OptSens_Distance
        absolute_velocity: OptSens_LVL
        angle: OptSens_Angle
        vl: OptSens_VL
        vt: OptSens_VT
    tire_mapping:
      frame_id: "base_link"
      canbus_message_fl_id: 577
      canbus_message_fr_id: 578
      canbus_message_rl_id: 833
      canbus_message_rr_id: 834
      field_mapping:
        FL:
          rpm: Inv_FL_Velocity_ActVal
          torq: Inv_FL_Torque_ActVal
        FR:
          rpm: Inv_FR_Velocity_ActVal
          torq: Inv_FR_Torque_ActVal
        RL:
          rpm: Inv_RL_Velocity_ActVal
          torq: Inv_RL_Torque_ActVal
        RR:
          rpm: Inv_RR_Velocity_ActVal
          torq: Inv_RR_Torque_ActVal

